### PR TITLE
enhance integration tests for nested subquery expressions

### PR DESCRIPTION
**Pull request by mikethebeer**
Enhances the tests for nested subquery expressions used with JOINs

### CODE REVIEW

File: sql/src/test/java/io/crate/integrationtests/SubSelectIntegrationTest.java [hunk 1]
```diff
@@ -588,4 +588,63 @@ public void testSubqueryExpressionWithInPredicateEvaluatesToNull() throws Except
...
+            is ("asok| internship\n" +
+                "catbert| HR\n" +
+                "dilbert| engineering\n" +
+                "pointy haired boss| management\n" +
+                "ratbert| HR\n" +
+                "wally| engineering\n"));
+    }
+
+    @Test
+    public void testSubselectWithJoin() throws Exception {
```
Comment (mxm) [inline sql/src/test/java/io/crate/integrationtests/SubSelectIntegrationTest.java]: How about `testSemiJoinWithNestedJoin`?

File: sql/src/test/java/io/crate/integrationtests/SubSelectIntegrationTest.java [hunk 2]
```diff
@@ -588,4 +588,63 @@ public void testSubqueryExpressionWithInPredicateEvaluatesToNull() throws Except
         assertThat(response.rowCount(), is(1L));
         assertNull(response.rows()[0][0]);
     }
+
+    @Test
+    public void testNestedSubqueryWithAggregatesInMultipleStages() throws Exception {
```
Comment (mxm) [inline sql/src/test/java/io/crate/integrationtests/SubSelectIntegrationTest.java]: How about `testNestedSemiJoinWithAggregatesInMultipleStages`?

Comment (matriv) [reply sql/src/test/java/io/crate/integrationtests/SubSelectIntegrationTest.java]: @mxm semi-joins not in master yet, this is just 2level multiphase-execution, or am I missing something?

File: sql/src/test/java/io/crate/integrationtests/SubSelectIntegrationTest.java (hunk 1 above)
Comment (matriv) [reply sql/src/test/java/io/crate/integrationtests/SubSelectIntegrationTest.java]: @mxm same here.

File: sql/src/test/java/io/crate/integrationtests/SubSelectIntegrationTest.java (hunk 2 above)
Comment (mikethebeer) [reply sql/src/test/java/io/crate/integrationtests/SubSelectIntegrationTest.java]: @mxm @matriv yes indeed. That's why I don't want to name it to semi-joins.

**Pull request merged**